packages/big-design/src/components/Timepicker/Timepicker.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { forwardRef, memo, Ref } from 'react';
+
+import { Select } from '../Select';
+
+interface Props {
+  error?: React.ReactNode;
+  label?: string;
+  onTimeChange(date: string): void;
+}
+
+interface PrivateProps {
+  forwardedRef: Ref<HTMLInputElement>;
+}
+
+export type TimepickerProps = Props & React.InputHTMLAttributes<HTMLInputElement>;
+
+function getTimeIntervals() {
+  const times = ['12:00 AM'];
+
+  for (let i = 1; i < 24; i++) {
+    i < 12 ? times.push(`${i}:00 AM`) : times.push(`${i === 12 ? i : i - 12}:00 PM`);
+  }
+  times.push('11:59 PM');
+
+  return times.map((time) => ({ value: time, content: time }));
+}
+
+const RawTimePicker: React.FC<TimepickerProps & PrivateProps> = ({
+  error,
+  forwardedRef,
+  label,
+  onTimeChange,
+  value,
+  ...props
+}) => {
+  return (
+    <Select
+      label={label}
+      error={error}
+      value={value}
+      onOptionChange={onTimeChange}
+      placeholder="hh : mm"
+      options={getTimeIntervals()}
+      inputRef={forwardedRef}
+      {...props}
+    />
```

Comment:
🤔 We might want a `StyleableSelect` component (for internal consumption only) if we want to get smaller `Select` width styles. @deini thoughts on this, or should a normal `Select` be fine?

Reply:
I'm not opposed to an internal `StyleableSelect`, however, is the size of the select an issue? Quoting @animesh1987:
>From what I have observed once we have the Datetimepicker combined into one then we will have the design outcome as per the figma file.

Seems like we are able to do what we want without any changes. It can also be wrapped in a container that controls the width. 🤔

Reply:
That's in regards to the width of the field. I should've clarified, my comment is in regards to the width of the dropdown:
<img width="104" alt="Screen Shot 2020-07-30 at 15 38 47" src="https://user-images.githubusercontent.com/10539418/88972198-cd464900-d27a-11ea-97fb-28e85f83a9d3.png">

Reply:
Gotcha, yeah a styleable internal one or adding a prop for it to be driven by content.

Reply:
We might want to consider that @animesh1987

Reply:
A prop driven by content is a better idea imo.

Reply:
Thanks for the feedback.

So to summarise it we are fine with adding a prop to existing select component?

Reply:
Maybe `useContentWidth` or `autoWidth`. We've been discussing a similar prop for buttons and we should be uniform on this matter. My card is in for `autoWidth`, but what does everyone else think?

Another thing to take not of is that `fit-content` doesn't have support in a lot of browsers, so I'd rather use `auto` which should work in all.

Otherwise that looks good.